youtube_dl/extractor/tvplay.py

```diff
@@ -321,6 +320,24 @@ def _real_extract(self, url):
                 'url': sami_path,
             }]
 
+        subtitles_webvtt = video.get('subtitles_webvtt')
+        if subtitles_webvtt:
+            lang = self._search_regex(
+                r'_([a-z]{2})\.vtt', subtitles_webvtt, 'lang',
+                default=compat_urlparse.urlparse(url).netloc.rsplit('.', 1)[-1])
+            subtitles[lang] = [{
+                'url': subtitles_webvtt,
+            }]
+
+        subtitles_for_hearing_impaired = video.get('subtitles_for_hearing_impaired')
+        if subtitles_for_hearing_impaired:
+            lang = self._search_regex(
+                r'_([a-z]{2})_', subtitles_for_hearing_impaired, 'lang',
+                default=compat_urlparse.urlparse(url).netloc.rsplit('.', 1)[-1])
+            subtitles[lang + '_sdh'] = [{
+                'url': subtitles_for_hearing_impaired,
+            }]
```

Comment:
Code duplication.

Reply:
Should I create a function and use it for sami_path, subtitles_webvtt and subtitles_for_hearing_impaired or what is the preferred way?

Reply:
Plain `for` is enough.